src/components/Assets/Catalog/index.tsx

```diff
@@ -99,9 +105,18 @@ export function EgeriaAssetCatalog() {
 
   useEffect(() => {
     const _queryParams = getQueryParams(searchParams);
-
+    let qip = form.qIsPristine
+    let tip = form.typesIsPristine
+    if (_queryParams.q !== '' && qip) {
+      qip = false;
+    }
+    if (!isArrayEmpty(_queryParams.types) && tip) {
+      tip = false;
+    }
     setForm({
-      ..._queryParams
+      ..._queryParams,
+      qIsPristine : qip,
+      typesIsPristine : tip
     });
```

Comment:
```js
setForm({
  q: {
    ...form.q,
    value: _queryParams.q || '',
    isValid: isStringLonger(_queryParams.q || '', QUERY_MIN_LENGTH)
  },
  types: {
    ...form.types,
    value: [..._queryParams.types],
    isValid: !isArrayEmpty([..._queryParams.types])
  },
  // i have to add the rest of the form fields manually as we cannot use the _queryParams
  // because we changed the data type on q and types to object<{value, isPristine}>
  pageSize: _queryParams.pageSize,
  exactMatch: _queryParams.exactMatch,
  caseSensitive: _queryParams.types.caseSensitive
});
```